python/looker_sdk/rtl/api_settings.py

```diff
@@ -32,13 +32,21 @@
 from looker_sdk.rtl import transport
 
 if sys.version_info >= (3, 8):
-    from typing import Protocol
+    from typing import Protocol, TypedDict
 else:
-    from typing_extensions import Protocol
+    from typing_extensions import Protocol, TypedDict
 
+from typing_extensions import Required
+
+class SettingsConfig(TypedDict, total=False):
+    client_id: Required[str]
```

Comment:
`Required` is a relatively recent addition to `typing_extensions`, looks like about 3 months ago.

pyright support is there which is nice..

![settings-config](https://user-images.githubusercontent.com/94563272/155230306-4bb95a6c-2796-4c0a-aa0f-b2475a7e02f0.png)

Reply:
cool! I'm approving action runs so we can see how this goes on all the python versions we support